File: pyg_parent/pyg_shop_web/src/main/java/com/pinyougou/shop/controller/GoodsController.java
package com.pinyougou.shop.controller;

import java.util.List;

import com.pyg.pyg_sellergoods_interface.GoodsService;
import com.pyg_pojo.entity.Goods;
import com.pyg_pojo.entity.PageResult;
import com.pyg_pojo.entity.Result;
import com.pyg_pojo.pojo.TbGoods;
import org.apache.activemq.ActiveMQXAConnectionFactory;
import org.apache.activemq.command.ActiveMQQueue;
import org.apache.activemq.command.ActiveMQTopic;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.jms.core.MessageCreator;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;

import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.ObjectMessage;
import javax.jms.Session;

@SuppressWarnings("all")
@RestController
@RequestMapping("/goods")
public class GoodsController {
    @Reference
    private GoodsService goodsService;
    /*
      @Reference
      private SeachItenService seachItenService;

      @Reference
      private FreeMarkerService freeMarkerService;*/
    @Autowired
    private JmsTemplate jmsTemplate;
    //添加sorl库数据
    @Autowired
    private ActiveMQQueue queueSolrCreateDestination;
    //删除solr库数据
    @Autowired
    private ActiveMQQueue queueSolrDeleteDestination;
    //生成静态模板
    @Autowired
    private ActiveMQTopic topicPageCreateDestination;
    //删除静态模板
    @Autowired
    private ActiveMQTopic topicPageDeleteDestination;

    @RequestMapping("/updateStatus")
    public Result updateStatus(Long[] ids, String status) {
        try {
            goodsService.updateStatus(ids, status);
            if ("4".equals(status)) {
                //上架
                //添加solr库数据
                //seachItenService.importItem(ids);
                //发送消息
                jmsTemplate.send(queueSolrCreateDestination, new MessageCreator() {
                    @Override
                    public Message createMessage(Session session) throws JMSException {
                        return session.createObjectMessage(ids);
                    }
                });

                //生成模板
                for (Long id : ids) {
                    //freeMarkerService.importTemplate(id);
                    jmsTemplate.send(topicPageCreateDestination, new MessageCreator() {
                        @Override
                        public Message createMessage(Session session) throws JMSException {
                            return session.createObjectMessage(id);
                        }
                    });
                }
            }
            if ("5".equals(status)) {
                //下架
                //删除solr库数据
                //seachItenService.removeItem(ids);
                jmsTemplate.send(queueSolrDeleteDestination, new MessageCreator() {
                    @Override
                    public Message createMessage(Session session) throws JMSException {
                        return session.createObjectMessage(ids);
                    }
                });
                //删除模板
                for (Long id : ids) {
                    //freeMarkerService.removeTemplate(id);
                    jmsTemplate.send(topicPageDeleteDestination, new MessageCreator() {
                        @Override
                        public Message createMessage(Session session) throws JMSException {
                            return session.createObjectMessage(id);
                        }
                    });
                }
            }
            if ("6".equals(status)) {
                //商家关闭
                //删除solr库数据
                //seachItenService.removeItem(ids);
                jmsTemplate.send(queueSolrDeleteDestination, new MessageCreator() {
                    @Override
                    public Message createMessage(Session session) throws JMSException {
                        return session.createObjectMessage(ids);
                    }
                });
                //删除模板
                for (Long id : ids) {
                    // freeMarkerService.removeTemplate(id);
                    jmsTemplate.send(topicPageDeleteDestination, new MessageCreator() {
                        @Override
                        public Message createMessage(Session session) throws JMSException {
                            return session.createObjectMessage(id);
                        }
                    });
                }
            }
            return new Result(true, "操作成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "操作失败");
        }
    }

    /**
     * 返回全部列表
     *
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbGoods> findAll() {
        return goodsService.findAll();
    }

    // 根据商品id查询
    @RequestMapping("/findOne")
    public Goods findOne(Long id) {
        return goodsService.findOne(id);
    }

    /**
     * 返回全部列表
     *
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page, int rows) {
        return goodsService.findPage(page, rows);
    }

    /**
     * 增加
     *
     * @param goods
     * @return
     */
    @RequestMapping("/add")
    public Result add(@RequestBody Goods goods) {
        try {
            //添加商家用户登录名
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            goods.getTbGoods().setSellerId(name);
            goodsService.add(goods);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "增加失败");
        }
    }

    // 修改商品信息
    @RequestMapping("/update")
    public Result update(@RequestBody Goods goods) {
        try {
            // 添加商家用户id
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            goods.getTbGoods().setSellerId(name);
            goodsService.update(goods);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            // TODO: handle exception
            return new Result(false, "修改失败");
        }
    }


    /**
     * 批量删除
     *
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            goodsService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }

    /**
     * 查询+分页
     *
     * @param
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbGoods goods, int page, int rows) {
        //商家只查询自己的商品信息
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        goods.setSellerId(name);
        return goodsService.findPage(goods, page, rows);
    }

}

File: pyg_parent/pyg_manager_web/src/main/webapp/js/controller/goodsController.js
 //控制层 
app.controller('goodsController' ,function($scope,$controller,itemCatService,goodsService){
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		goodsService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		goodsService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		goodsService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var serviceObject;//服务层对象  				
		if($scope.entity.id!=null){//如果有ID
			serviceObject=goodsService.update( $scope.entity ); //修改  
		}else{
			serviceObject=goodsService.add( $scope.entity);//增加
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					//重新查询 
		        	$scope.reloadList();//重新加载
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框			
		goodsService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		goodsService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
    $scope.status = ["未申请", "申请中", "审核通过", "审核未通过","商品上架","商品下架","商品关闭"];
    //查询商品分类信息
    $scope.itemList=[];
    $scope.findAll = function () {
        itemCatService.findAll().success(
            function (respose) {
                for (var i = 0; i <respose.length;i++) {
                    $scope.itemList[respose[i].id]=respose[i].name;

                }
            }
        );
    }

    //修改商品状态信息
    $scope.updateStatus=function(status){
    	alert($scope.selectIds);
        goodsService.updateStatus($scope.selectIds,status).success(
            function (respose) {
                if (respose.success){
                    $scope.reloadList();
                    $scope.selectIds=[];
                } else{
                    alert("操作失败");
                }
            }
        );
    }
});	
